src/transform/src/redundant_join.rs

```diff
@@ -103,9 +107,31 @@ impl RedundantJoin {
                     Ok(result)
                 }
 
-                MirRelationExpr::LetRec { .. } => {
-                    // TODO
-                    Err(crate::TransformError::LetRecUnsupported)?
+                MirRelationExpr::LetRec { ids, values, body } => {
+                    // As a first approximation, we extend the `lets` context as follows.
+                    for (id, value) in zip_eq(ids.iter(), values.iter()) {
+                        let val_type = value.typ();
+                        let prov_new = ProvInfo::make_leaf(Id::Local(*id), val_type.arity());
+                        let prov_old = lets.insert(Id::Local(*id), vec![prov_new]);
+                        assert!(prov_old.is_none());
+                    }
```

Comment:
It's a bit more explicit. I think the `MirRelationExpr::Get` is primarily there to cover `GlobalId`. I can rely on it not differentiating between `LocalId` and `GlobalId` at the moment, but I would say that this is incidental.

All other `Let` / `LetRec` handling code throughout the `transform` crate is enriching the context before descending to the `value` / `body` sub-terms, so I think it will be weird to leave it out from here.